code_generation/license.h

```diff
@@ -83,6 +87,27 @@ class KODE_EXPORT License
      */
     QString text() const;
 
+    /**
+     * Returns license Type enum element by name. If there no such license,
+     * NoLicense will be returned
+     * @param license_string Name of license
+     * @return one of Type elements
+     */
+    static Type getLicenseNameByString(const QString &license_string);
+
+    /**
+     * Return license name by Type enum element.
+     * @param license_enum one of Type elements
+     * @return name of license
+     */
+    static QString getLicenseNameByEnum(const Type &license_enum);
```

Comment:
Wouldn't it be better object-oriented design to make this one non-static, like

    QString typeName() const

which means, for the calling side, if you don't have a License object yet, the equivalent to getLicenseNameByEnum(License::GPL) would be License(License::GPL).typeName(). No big deal IMHO. And if you do have a License instance then it will read much nicer.